package-res/resources/web/pentaho/visual/base/View.js

```diff
@@ -92,39 +89,105 @@ define([
     },
 
     /**
-     * Orchestrates the rendering of the visualization.
+     * Orchestrates the rendering of the visualization and is meant to be invoked by the container.
      *
-     * This method executes [_render]{@link pentaho.visual.base.View#_render}
-     * asynchronously and is meant to be invoked by the container.
+     * Executes [_update]{@link pentaho.visual.base.View#_update} asynchronously in
+     * the will/did/rejected event loop associated with an update of a visualization,
+     * and also creates the visualization DOM element the first time it successfully updates.
+     *
+     * In order to get the visualization DOM element,
+     * listen to the {@link pentaho.visual.base.events.DidCreate|"did:create"} event.
      *
      * @return {Promise} A promise that is fulfilled when the visualization
      * is completely rendered. If the visualization is in an invalid state, the promise
      * is immediately rejected.
+     *
+     * @fires "will:update"
+     * @fires "did:update"
+     * @fires "rejected:update"
+     * @fires "did:create"
      */
-    render: function() {
-      var me = this;
+    update: function() {
```

Comment:
Please create an internal field `_updatingPromise` and a corresponding `isUpdating` boolean getter and keep the field in sync within this method.

Calling `update` while the view is already updating is not guaranteed to be the same as simply waiting for the current update to finish, as the new call may be due to subsequent model changes not fully captured by the current `update` call.

So, at the top of the method, if the view is already updating, the call should  return a rejected Promise with some specific UserError.
I think it should not trigger update related events in this case.
An alternative would be to schedule an update call for when the current update finishes (by waiting on the current promise), but I'm not sure that would be the common desire.